src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -416,6 +442,66 @@ VST3Wrapper::VST3Wrapper(VST3::Hosting::Module& module, const VST3::Hosting::Cla
    mEditController = editController;
 
    mEditController->initialize(&AudacityVst3HostApplication::Get());
+
+   if(GetCache(mEffectClassInfo.ID()) == nullptr)
+   {
+      //First time instantiation...
+      auto& cache = CreateCache(mEffectClassInfo.ID());
+
+      //Find root unit program change parameter ID (if any)
+      for(int32 parameterIndex = 0, parametersCount = mEditController->getParameterCount();
+         parameterIndex < parametersCount;
+         ++parameterIndex)
+      {
+         Vst::ParameterInfo parameterInfo {};
+         if(mEditController->getParameterInfo(parameterIndex, parameterInfo) != kResultOk)
+            continue;
+
+         if(parameterInfo.unitId != Vst::kRootUnitId)
+            continue;
+         if((parameterInfo.flags & Vst::ParameterInfo::kIsProgramChange) == 0)
+            continue;
+
+         cache.rootUnitProgramChangeParameterID = parameterInfo.id;
```

Comment:
... and I think this assignment makes that distinguishability.

Reply:
Yes, the meaning for `kNoProgramListId` is that plugin does not have program list parameter, which is fine.